net_socket: mimxrt1170_evk_cm7: build failure

**Describe the bug**
all ethernet driver include application build fail on rt11xx platforms

Please also mention any information which could help others to understand
the problem you're facing:
```
/home/shared/disk/zephyr_project/zephyr_test/zephyr/drivers/ethernet/eth_mcux.c:982:7: error: incompatible type for argument 4 of 'handle->callback'
  982 |       NULL, handle->userData);
      |       ^~~~
      |       |
      |       void *
/home/shared/disk/zephyr_project/zephyr_test/zephyr/drivers/ethernet/eth_mcux.c:982:7: note: expected 'enet_event_t' {aka 'enum _enet_event'} but argument is of type 'void *'
/home/shared/disk/zephyr_project/zephyr_test/zephyr/drivers/ethernet/eth_mcux.c:980:6: error: too few arguments to function 'handle->callback'
  980 |      handle->callback(context->base,
      |      ^~~~~~
[145/307] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/ipv6_nbr.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /home/ubuntu/.local/bin/cmake --build /home/shared/disk/zephyr_project/zephyr_test/zephyr/tests/net/socket/tcp/build

```
 - ...

**To Reproduce**
Steps to reproduce the behavior:
cd tests/net/socket/tcp
1. mkdir build; cd build
2. cmake -DBOARD=mimxrt1170_evk_cm7 ..
3. make
4. See error

**Expected behavior**
build and run pass

**Impact**
all ethernet applications

**Logs and console output**
```
/home/shared/disk/zephyr_project/zephyr_test/zephyr/drivers/ethernet/eth_mcux.c: In function 'eth_tx_thread':
/home/shared/disk/zephyr_project/zephyr_test/zephyr/drivers/ethernet/eth_mcux.c:982:7: error: incompatible type for argument 4 of 'hanle->callback'
  982 |       NULL, handle->userData);
      |       ^~~~
      |       |
      |       void *
/home/shared/disk/zephyr_project/zephyr_test/zephyr/drivers/ethernet/eth_mcux.c:982:7: note: expected 'enet_event_t' {aka 'enum _enet_vent'} but argument is of type 'void *'
/home/shared/disk/zephyr_project/zephyr_test/zephyr/drivers/ethernet/eth_mcux.c:980:6: error: too few arguments to function 'handle->cllback'
  980 |      handle->callback(context->base,
      |      ^~~~~~
[145/307] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/ipv6_nbr.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /home/ubuntu/.local/bin/cmake --build /home/shared/disk/zephyr_project/zephyr_test/zephyr/tsts/net/socket/tcp/build


```
**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v3.0.0-rc3

--- a/drivers/ethernet/eth_mcux.c
+++ b/drivers/ethernet/eth_mcux.c
@@ -977,9 +977,15 @@ static void eth_tx_thread(void *arg1, void *unused1, void *unused2)
 				enet_handle_t *handle = &context->enet_handle;
 
 				if (handle->callback != NULL) {
+#if FSL_FEATURE_ENET_QUEUE > 1
+					handle->callback(context->base,
+						handle, 0, kENET_TxEvent,
+						NULL, handle->userData);
+#else
 					handle->callback(context->base,
 						handle, kENET_TxEvent,
 						NULL, handle->userData);
+#endif
 				}
 			}
 			ENET_EnableInterrupts(context->base,

